src/lib/address/ens.ts

```diff
@@ -0,0 +1,68 @@
+// @ts-ignore TODO: add type declarations for ensjs module
+import ENS, { getEnsAddress } from '@ensdomains/ensjs'
+import { ChainAdapterManager } from '@shapeshiftoss/chain-adapters'
+import { ChainTypes } from '@shapeshiftoss/types'
+import { getConfig } from 'config'
+import memoize from 'lodash/memoize'
+import { ethChainId } from 'test/mocks/accounts'
+import { getChainAdapters } from 'context/PluginProvider/PluginProvider'
+import { getWeb3Provider } from 'lib/web3-provider'
+
+import { ReverseLookupVanityDomain } from './address'
+import { ResolveVanityDomain, ResolveVanityDomainReturn, ValidateVanityDomain } from './address'
+
+let makeEns: () => void
+// getEnsAddress takes a magic number as string, networkId. 1 stands for mainnet
+const ens = new Promise<void>(resolve => (makeEns = resolve)).then(async () => {
+  const unchainedUrls = {
+    ethereum: {
+      httpUrl: getConfig().REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL,
+      wsUrl: getConfig().REACT_APP_UNCHAINED_ETHEREUM_WS_URL,
+    },
+  }
+
+  const ethereumChainAdapter = new ChainAdapterManager(unchainedUrls).byChain(ChainTypes.Ethereum)
+  const chainId = ethereumChainAdapter.getChainId()
```

Comment:
Do we still need this? We can probably use the exported `ethChainId` now, which means we won't have to instantiate `ethereumChainAdapter` here. I'd still keep the `chainIdReference` var as it makes it clear why/how we're getting the magic 1

Reply:
this is a mess yes, i'll fix, and make this whole getter synchronous and sane